### PR TITLE
Storage macOS Quickstart SwiftUI update

### PR DESCRIPTION
**Pull request by granluo**

### CODE REVIEW

File: storage/StorageExample/Shared/ContentView.swift
Comment (morganchen12) [inline storage/StorageExample/Shared/ContentView.swift:24]: I'd recommend not doing this since it's clever, but not extensible: If the app ever depends on an iOS or macOS specific image feature, the typealias will break. Instead, you could wrap this in an `Image` helper struct that references the correct platform-specific type below and then redefine methods on top of it.

File: storage/StorageExample/Shared/ContentView.swift
Comment (morganchen12) [inline storage/StorageExample/Shared/ContentView.swift:284]: Instead of force-unwrapping in this method, you can just return `nil` since the return type is optional.

File: storage/StorageExample/Shared/ImagePicker.swift
Comment (morganchen12) [inline storage/StorageExample/Shared/ImagePicker.swift:108]: This should be passed to the continuation.

File: storage/StorageExample/Shared/ImagePicker.swift
Comment (morganchen12) [inline storage/StorageExample/Shared/ImagePicker.swift:129]: This error should be rethrown.

File: storage/StorageExample/Shared/ImagePicker.swift
Comment (morganchen12) [inline storage/StorageExample/Shared/ImagePicker.swift:114]: This should be rethrown (you can likely just omit the top-level do/catch).

File: storage/StorageExample/Shared/ImagePicker.swift
Comment (morganchen12) [inline storage/StorageExample/Shared/ImagePicker.swift:144]: Use `if let image = self.image` to avoid the force-unwrap.

File: storage/StorageExample/Shared/ImagePicker.swift
Comment (morganchen12) [inline storage/StorageExample/Shared/ImagePicker.swift:170]: nit: `url` instead of `urll`

File: storage/StorageExample/Shared/ImagePicker.swift
Comment (morganchen12) [inline storage/StorageExample/Shared/ImagePicker.swift:130]: `throw` this error (here and throughout) instead of returning nil.

**Review by morganchen12 — COMMENTED** [review]
LGTM with one last comment. Please wait for @ncooke3's approval before merging.

File: storage/StorageExample/Shared/ImagePicker.swift
Comment (morganchen12) [inline storage/StorageExample/Shared/ImagePicker.swift:121]: Can you not just `throw error` instead of defining a separate error enum?

**Review by morganchen12 — APPROVED** [review]
LGTM

File: storage/StorageExample/Shared/ContentView.swift
Comment (ncooke3) [inline storage/StorageExample/Shared/ContentView.swift:105]: nit: remove new line 
```suggestion
```

File: storage/StorageExample/Shared/ContentView.swift
Comment (ncooke3) [inline storage/StorageExample/Shared/ContentView.swift:269]: optional: Does this work? If so, it avoids the need for duplicate `setImage(fromURL url: URL) -> Image` APIs.

```suggestion
  #if os(iOS)
    func setImage(fromImage image: UIImage) -> Image {
      return Image(uiImage: image)
    }
  #elseif os(macOS)
    func setImage(fromImage image: NSImage) -> Image {
      return Image(nsImage: image)
    }
  #endif

  func setImage(fromURL url: URL) -> Image {
    return setImage(fromImage: .init(contentsOfFile: url.path)!)
  }

```

File: storage/StorageExample/Shared/ContentView.swift
Comment (ncooke3) [inline storage/StorageExample/Shared/ContentView.swift:302]: nit: remove newline
```suggestion
```

File: storage/StorageExample/Shared/ViewModel.swift
Comment (ncooke3) [inline storage/StorageExample/Shared/ViewModel.swift:31]: Do you need to `import AppKit` if `os(macOS)`?

File: storage/StorageExample/Shared/ImagePicker.swift
Comment (ncooke3) [inline storage/StorageExample/Shared/ImagePicker.swift:170]: Curious, does an async/await API generate for this?
```swift
try await item.loadItem( //...
```

**Comment by ncooke3** [comment]
Thanks @granluo, looks great!

File: storage/StorageExample/Shared/ImagePicker.swift
Comment (granluo) [reply storage/StorageExample/Shared/ImagePicker.swift:170]: It looks like [loadItem](https://developer.apple.com/documentation/foundation/nsitemprovider/1403900-loaditem) does get an async note in its doc. Will update it.

File: storage/StorageExample/Shared/ViewModel.swift
Comment (granluo) [reply storage/StorageExample/Shared/ViewModel.swift:31]: I think `SwiftUI` is sufficient to handle the UI. Without importing `AppKit`, the app can be built and run successfully.

File: storage/StorageExample/Shared/ContentView.swift
Comment (granluo) [reply storage/StorageExample/Shared/ContentView.swift]: That's neat! It is working :)

Comment (ncooke3) [reply storage/StorageExample/Shared/ContentView.swift]: oh, cool!

**Review by ncooke3 — APPROVED** [review]
LGTM. nice work!

**Pull request merged**
